### ISSUE TITLE
Consider to use gradient descent or similar methods for searching optimal hyperparameters

### ISSUE THREAD

**Issue by deil87** [deil87/automl-genetic#172]
Right now we use evolutionary approach even at this level. We can do gradient search as well. We don't want to stick to grid values... we want to be able to converge
